### DISCUSSION TITLE
Jacobian not recorded correctly in AD preaccumulation region

### DISCUSSION THREAD

**Discussion by joshkellyjak** [su2code/SU2#2370] [General]
**Describe the bug**

You will have to bear with me, I might get some of this wrong, I am not entirely sure what is going on here so I will try and include as much information as possible. Using the tape tagging method developed by @oleburghardt  in #2343 and presented at the most recent SU2 Conference, a run of the full tape reveals that a variable used in a preaccumulation output is not properly recorded. The variable is the `val` in the function `PassiveAssign` in `CSysMatrix.hpp` line 215.

`FORCEINLINE static ScalarType PassiveAssign(const SrcType& val) {
    return SU2_TYPE::GetValue(val);
  }`

Following the call stack of the error reveals that the problem variables in `val` are the jacobian components calculated in the Roe and JST scheme (these are the two options I have tested). It is possible to resolve this error by simply moving the stopping of the preaccumulation region from before the call to `updateLinearSystem` in the `ComputeFlux` method of both these schemes, to after, as show below.

`/*--- Update the vector and system matrix. ---*/

    updateLinearSystem(iEdge, iPoint, jPoint, implicit, updateType,
                       updateMask, flux, jac_i, jac_j, vector, matrix);

    /*--- Stop preaccumulation. ---*/

    stopPreacc(flux);`

However, whilst this resolve the error in this location, the next error appears in the Giles boundary condition. Here the two jacobian components within the residual object are not recorded correctly in the preaccumulation output region also. This in `CEulerSolver.cpp:6556`.

`/*--- Compute the residual using an upwind scheme ---*/
      auto residual = conv_numerics->ComputeResidual(config);

      /*--- Update residual value ---*/
      LinSysRes.AddBlock(iPoint, residual);`

I have tried simpy adding these components as preacc outputs however they are SIMD arrays and the datatype cannot be converted using the existing implementation. 

I am not sure what is going on here, or the next steps to resolve this problem. Any further assistance would be greatly appreciated, thank you :)

**Bug report checklist**
*Please make sure that you have followed the checklist below, many common problems can be solved by:*

- [X] Consulting the build instructions (https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/).
- [X] Looking for similar problems on GitHub or CFD-Online (https://www.cfd-online.com/Forums/su2/).
- [X] Updating to the newest version of SU2 (either develop, master, or the pre-built executables https://su2code.github.io/download.html).

**Desktop (please complete the following information):**
 - OS: [MacOS Sonoma 14.6.1]
 - C++ compiler and version: [clang 14.0.3 "Apple clang version 14.0.3 (clang-1403.0.22.14.1)]
 - MPI implementation and version: [OpenMPI 5.0.3]
 - SU2 Version: [v8.0.1]

**Comment by pcarruscag** [comment]
The Jacobian matrix is passive, you'd need a TB of ram to run an inviscid NACA0012 otherwise.

**Comment by joshkellyjak** [comment]
Thank you, In this case what could be causing the tagging tool to interpret the jacobian as a preacc output? Maybe this is a bug in the tagging @oleburghardt ?

**Comment by Rteg54** [comment]
Hi Community,
i have done a mesh using ICEM Ansys, and i have the mesh file as cgns now and i want to change it to su2 format and i wrote a converting cfg file for thatbut when i try to run the simulation using su2 code it says error.
The case i am working with is a concentric cylinder coutte flow 3d. As i have basically a hollow cylinder where the inner cylinder rotates and the outer cylinder is fixed the way i named my boundary conditions in ICEM is inner_wall for the inner cylinder then outer_wall for outer cylinder then front is inlet and back is outlet. then i wrote the convert fill as the following:
% SET BOUNDARY MARKERS ACCORDING TO YOUR ICEMCFD PROJECT
MARKER_INLET= ( "INLET" ) % Inlet boundary
MARKER_OUTLET= ( "OUTLET" ) % Outlet boundary
MARKER_WALL_FUNCTIONS= ( "OUTER_WALL", "INNER_WALL" ) % Viscous walls

% MESH INPUT FILE
MESH_FILENAME= T1.cgns % Input CGNS file

% MESH INPUT FILE FORMAT (SU2, CGNS, NETCDF_ASCII)
MESH_FORMAT= CGNS % Format of the input mesh

% MESH OUTPUT FILE
MESH_OUT_FILENAME= T1.su2 % Output SU2 file

% SOME DUMMY PARAMETERS
DV_KIND= NO_DEFORMATION
DV_PARAM= ( 1.0 )
i would really appreciate the help.
